### PR TITLE
Feature/recipes cooking steps

### PR DESCRIPTION
**Pull request by itsandeepsingh**
Please review :)

### CODE REVIEW

File: components/CloseButton/CloseButton.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    opacity: 1;
+  }
+
+  @media (min-width: 500px) {
+    font-size: 25px;
+    margin: 8px 0;
+  }
+`;
+
+export default function CloseButton() {
```
Comment (tejue) [inline components/CloseButton/CloseButton.js:29]: If it fits with your App you could make this button more reusable. You style it here but give the functionallity an the text down as a prop. But that would only works if you want the apperance to be the same for the buttons.

File: components/CookingButton/CookingButton.js [hunk 2]
```diff
@@ -34,11 +35,23 @@ const Button = styled.button`
...
+      setCurrentStepIndex(0);
+    }
+  };
+
   return (
     <Container>
-      <Link href={`${1}`} passHref legacyBehavior>
-        <Button type="button">Jetzt kochen</Button>
-      </Link>
+      <Button type="button" onClick={handleStartCooking}>
```
Comment (tejue) [inline components/CookingButton/CookingButton.js:52]: see above :)

File: components/Recipe/Recipe.js [hunk 3]
```diff
@@ -69,7 +69,13 @@ export default function Recipe({
           onClick={onToggleSelection}
           aria-label={isSelected ? "unselect" : "select"}
```
Comment (tejue) [inline components/Recipe/Recipe.js:70]: You could be a bit more specific for the aria-label: what to select, what to unselect. That might be a bit more understandable by a screenreader

File: components/SelectionHook/SelectionHook.js [hunk 4]
```diff
@@ -16,13 +17,20 @@ const Hook = styled(Image)`
...
 
-export default function SelectionHook() {
+export default function SelectionHook({
+  onToggleSelection,
+  isSelected,
+  isSelectionMode,
+}) {
   return (
     <Hook
+      $isSelectionMode={isSelectionMode}
```
Comment (tejue) [inline components/SelectionHook/SelectionHook.js:27]: I don't know this $ in this position. What is it doing?

File: components/SelectionHook/SelectionHook.js [hunk 5]
```diff
@@ -16,13 +17,20 @@ const Hook = styled(Image)`
...
       src={require(`/assets/hook.png`).default}
       width={22}
       height={22}
       alt=""
+      onClick={onToggleSelection}
+      aria-label={isSelected ? "unselect" : "select"}
```
Comment (tejue) [inline components/SelectionHook/SelectionHook.js:33]: Same with aria-label above :)

File: lib/recipes.js [hunk 6]
```diff
@@ -1,3 +1,34 @@
+class RecipeStep {
```
Comment (tejue) [inline lib/recipes.js:1]: unfortunatelly I don't get this lib.js at all :/ I hope it works but I just don't understand the code

File: pages/index.js [hunk 7]
```diff
@@ -1,36 +1,13 @@
...
 
-export default function RecipesPage({
-  isSelectionMode,
-  handleToggleSelectionMode,
-  recipesSelection,
-  onToggleSelection,
-  selectedRecipesCount,
-  totalCookingTime,
-}) {
+export default function RecipesPage({}) {
```
Comment (tejue) [inline pages/index.js:5]: if you dont need to destructure you can delete {}

**Review by tejue — APPROVED** [review]
Hey Sandeep, it's a lot of code and I tried to get through it. I didn't understand everything but left you some comments. I'll approve because most of the comments are just small things you can think about or adapt.

File: components/SelectionHook/SelectionHook.js (hunk 4 above)
Comment (itsandeepsingh) [reply components/SelectionHook/SelectionHook.js:27]: without an $ I did receive this error. 

"Warning: React does not recognize the `isSelectionMode` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isselectionmode` instead. If you accidentally passed it from a parent component, remove it from the DOM element."

File: components/SelectionHook/SelectionHook.js (hunk 5 above)
Comment (itsandeepsingh) [reply components/SelectionHook/SelectionHook.js:33]: good point.

File: components/Recipe/Recipe.js (hunk 3 above)
Comment (itsandeepsingh) [reply components/Recipe/Recipe.js:70]: good point.

File: lib/recipes.js (hunk 6 above)
Comment (itsandeepsingh) [reply lib/recipes.js:1]: First I defined a class for recipe steps and a class for a recipe. A recipe can consist of several recipe steps. I created a local example data and I then put these example recipes into a list. And yes it works absolutely fine :)

File: pages/index.js (hunk 7 above)
Comment (itsandeepsingh) [reply pages/index.js:5]: i will change that in the next story.

**Pull request merged**